securesystemslib/signer/_key.py

```diff
@@ -196,6 +203,32 @@ def verify_signature(self, signature: Signature, data: bytes) -> None:
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def get_hash_algorithm_str(self) -> Any:
+        """Returns payload hash algorithm used for this key as a str
+
+        Raises:
+            UnsupportedAlgorithmError: if key type not suported
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_hash_algorithm(self) -> Any:
+        """Returns payload hash algorithm used for this key as a HashAlgorithm"""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_padding_name_str(self) -> Any:
+        """Return payload padding name used for this key as a str"""
+
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_padding_name(self, hash_algorithm: Any) -> Any:
+        """Return payload padding name used for this key as a AsymmetricPadding"""
+
+        raise NotImplementedError
+
```

Comment:
I can't really see a benefit of adding these to the Key interface. The only subclass, which implements them, is SSlibKey. Why don't make these SSlibKey-only methods?

Reply:
In `_azure_signer.py` and `_gcp_signer.py` both `AzureSigner` and `GCPSigner` take in a public key of type `Key` on which they use the get hash method. Can this be changed to SSlibKey?

Reply:
Hm. Good question. It looks like all `Signer` implementations take `Key`s, even though they can only handle specific Key implementations. That's probably okay. And I definitely don't want to disrupt anyone, by changing it.

But adding new behaviour to an abstract base class, which is specific to one subclass only, and raising NotImplementedError everywhere else, does not feel right.

Would it be okay to add regular functions, e.g. to _utils? 

 Thoughts, @jku?

Reply:
I admit I didn't look at the specific uses in AzureSigner and GCPSigner yet but I would say it's fine for them to handle SSlibKey only -- that sounds correct to me, what else could the key be?

Whether that should be done in practice by type checking inside the AzureSigner and GCPSigner functions or by changing some public argument types I can't say without having a closer look...

Reply:
In my last commit the methods are now SSlibKey-only methods. I added the stricter type checking on GCPSigner and AzureSigner and some of their methods including a method from the Signer class. It seems to pass all the tests, but a more detailed review by @lukpueh and @jku would be advised.

Reply:
Looks great! If the change isn't likely to bother any user, then I'd go with it. I'll give a detailed review after #836 is merged.